proposals/nnnn-member-macro-conformances.md

```diff
@@ -0,0 +1,116 @@
+# Member Macro Conformances
+
+* Proposal: [SE-NNNN](NNNN-filename.md)
+* Authors: [Doug Gregor](https://github.com/DougGregor)
+* Review Manager: TBD
+* Status: **Awaiting review**
+* Vision: [Macros](https://github.com/apple/swift-evolution/blob/main/visions/macros.md)
+* Implementation: [apple/swift#67758](https://github.com/apple/swift/pull/67758)
+* Review: ([pitch](https://forums.swift.org/t/pitch-member-macros-that-know-what-conformances-are-missing/66590))
+
+## Introduction
+
+The move from conformance macros to extension macros in [SE-0402](https://github.com/apple/swift-evolution/blob/main/proposals/0402-extension-macros.md) included the ability for extension macros to learn about which protocols the type already conformed to (e.g., because a superclass conformed or an explicit conformance was stated somewhere), so that the macro could avoid adding declarations and conformances that aren't needed. It also meant that any new declarations added are part of an extension---not the original type definition---which is generally beneficial, because it means that (e.g.) a new initializer doesn't suppress the memberwise initializer. It's also usually considered good form to split protocol conformances out into their own extensions.
+
+However, there are some times when the member used for the conformance really needs to be part of the original type definition. For example:
+
+- An initializer in a non-final class needs to be a `required init` to satisfy a protocol requirement.
+- An overridable member of a non-final class.
+- A stored property or case can only be in the primary type definition.
+
+For these cases, a member macro can produce the declarations. However, member macros aren't provided with any information about which protocol conformances they should provide members for, so a macro might erroneously try to add conforming members to a type that already conforms to the protocol (e.g., through a superclass). This can make certain macros---such as macros that implement the `Encodable` or `Decodable` protocols---unimplemented.
+
+## Proposed solution
+
+To make it possible for a member macro to provide the right members for a conformance, we propose to extend member macros with the same ability that extension macros have to reason about conformances. Specifically:
+
+* The `attached` attribute specifying a `member` role gains the ability to specify the set of protocol conformances it is interest in, the same way an `extension` macro specifies the conformances it can provide.
```

Comment:
```suggestion
* The `attached` attribute specifying a `member` role gains the ability to specify the set of protocol conformances it is interested in, the same way an `extension` macro specifies the conformances it can provide.
```